src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java

```diff
@@ -53,6 +54,42 @@ public void before() throws Exception {
         rootLocation = "build/resources/test/reader/";
     }
 
+    @Test
+    public void testShardBulkParseAndQuery() throws Exception {
+        deleteAll();
+        ReaderMetricsProcessor mp =
+                new ReaderMetricsProcessor(rootLocation, true, new AppContext());
+
+        mp.processMetrics(rootLocation, 1566413985000L);
+        mp.processMetrics(rootLocation, 1566413990000L);
+
+        Result<Record> res =
+                mp.getMetricsDB()
+                        .getValue()
+                        .queryMetric(
+                                Arrays.asList("ShardEvents", "ShardBulkDocs"),
+                                Arrays.asList("sum", "sum"),
+                                Arrays.asList("Operation"));
+
+        boolean shardbulkEncountered = false;
+
+        for (Record record : res) {
+            if (PerformanceAnalyzerMetrics.sShardBulkPath.equals(record.get("Operation"))) {
+                assertNotNull(record.get("ShardEvents"));
+                assertNotNull(record.get("ShardBulkDocs"));
+                assertTrue((Double) record.get("ShardEvents") >= 1.0);
+                assertTrue((Double) record.get("ShardBulkDocs") >= 1.0);
```

Comment:
Asserting for `>= 1.0` checks for missing data but not for the accuracy. Let us assert for value of these `ShardEvents` and `ShardBulkDocs`

Can use [AdmissionControlSnapshotTest](https://github.com/opensearch-project/performance-analyzer-rca/blob/4734d6d5ba68ba41b6c62ab2cbf5ff22aa52a186/src/test/java/org/opensearch/performanceanalyzer/reader/AdmissionControlSnapshotTest.java) as reference

Reply:
Agreed, I've put the exact expected values.